### PR TITLE
Add waitForStringInLog for SRVE0242I (/metrics initialization)

### LINKED ISSUE #12589: Test Failure: com.ibm.ws.logging.fat.CustomAccessLogFieldsTest.testFieldsInAccessLogAreSameInJSON
RTC 275245

```
testFieldsInAccessLogAreSameInJSON:java.lang.NullPointerException: 2020-06-11-13:33:49:132 null
at com.ibm.ws.logging.fat.CustomAccessLogFieldsTest.parseIntoKvp(CustomAccessLogFieldsTest.java:662)
at com.ibm.ws.logging.fat.CustomAccessLogFieldsTest.testFieldsInAccessLogAreSameInJSON(CustomAccessLogFieldsTest.java:317)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at componenttest.custom.junit.runner.FATRunner$1.evaluate(FATRunner.java:197)
at componenttest.custom.junit.runner.FATRunner$2.evaluate(FATRunner.java:318)
at componenttest.custom.junit.runner.FATRunner.run(FATRunner.java:171)
```

RTC 275272

```
testChangeJsonAccessLogFieldsConfigValue:java.lang.NullPointerException: 2020-06-12-10:02:22:390 null
at com.ibm.ws.logging.fat.CustomAccessLogFieldsTest.areFieldsPresent(CustomAccessLogFieldsTest.java:601)
at com.ibm.ws.logging.fat.CustomAccessLogFieldsTest.testChangeJsonAccessLogFieldsConfigValue(CustomAccessLogFieldsTest.java:355)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at componenttest.custom.junit.runner.FATRunner$1.evaluate(FATRunner.java:197)
at componenttest.custom.junit.runner.FATRunner$2.evaluate(FATRunner.java:318)
at componenttest.custom.junit.runner.FATRunner.run(FATRunner.java:171)
```

### PR DESCRIPTION
**Pull request by jennifer-c**
#build

Adding a wait for the /metrics endpoint to be initialized before attempting to reach it.

Fixes #12589

### CODE REVIEW

**Comment by jennifer-c** [comment]
Ran build on Sun EBC where test was failing consistently. FAT ran 10 times.

https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_GGgpkLezEeqI3q2xeI2nKA

File: dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/CustomAccessLogFieldsTest.java [hunk 1]
```diff
@@ -169,6 +169,12 @@ public void cleanUp() throws Exception {
     public void testAccessLogFieldNamesEnv() throws Exception {
         setUp(envServer);
         waitForSecurityPrerequisites(envServer, WAIT_TIMEOUT);
+        // Ensure defaultHttpEndpoint-ssl TCP Channel is started
+        assertNotNull("TCP Channel defaultHttpEndpoint-ssl has not started (CWWKO0219I not found)",
+                      envServer.waitForStringInLog("CWWKO0219I.*defaultHttpEndpoint-ssl", 60000));
+        // Wait for /metrics to be initialized
+        assertNotNull("/metrics was not initialized (SRVE0242I not found)",
+                      envServer.waitForStringInLog("SRVE0242I", WAIT_TIMEOUT));
```
Comment (fmhwong) [inline dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/CustomAccessLogFieldsTest.java:177]: Use  "SRVE0242I.*/metrics.*" to be more specific for metrics endpoint.

File: dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/CustomAccessLogFieldsTest.java [hunk 2]
```diff
@@ -184,6 +190,12 @@ public void testAccessLogFieldNamesEnv() throws Exception {
     public void testAccessLogFieldNamesBootstrap() throws Exception {
         setUp(bootstrapServer);
         waitForSecurityPrerequisites(bootstrapServer, WAIT_TIMEOUT);
+        // Ensure defaultHttpEndpoint-ssl TCP Channel is started
+        assertNotNull("TCP Channel defaultHttpEndpoint-ssl has not started (CWWKO0219I not found)",
+                      bootstrapServer.waitForStringInLog("CWWKO0219I.*defaultHttpEndpoint-ssl", 60000));
+        // Wait for /metrics to be initialized
+        assertNotNull("/metrics was not initialized (SRVE0242I not found)",
```
Comment (fmhwong) [inline dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/CustomAccessLogFieldsTest.java:197]: Use  "SRVE0242I.*/metrics.*" to be more specific for metrics endpoint.

File: dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/CustomAccessLogFieldsTest.java [hunk 3]
```diff
@@ -200,10 +212,14 @@ public void testAccessLogFieldNamesXml() throws Exception {
...
-        } else {
-            assertNotNull("TCP Channel defaultHttpEndpoint-ssl has not started (CWWKO0219I not found)",
-                          xmlServer.waitForStringInLog("CWWKO0219I.*defaultHttpEndpoint-ssl", 60000));
         }
+        // Ensure defaultHttpEndpoint-ssl TCP Channel is started
+        assertNotNull("TCP Channel defaultHttpEndpoint-ssl has not started (CWWKO0219I not found)",
+                      xmlServer.waitForStringInLog("CWWKO0219I.*defaultHttpEndpoint-ssl", 60000));
+        // Wait for /metrics to be initialized
+        assertNotNull("/metrics was not initialized (SRVE0242I not found)",
+                      xmlServer.waitForStringInLog("SRVE0242I", WAIT_TIMEOUT));
```
Comment (fmhwong) [inline dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/CustomAccessLogFieldsTest.java:221]: Use  "SRVE0242I.*/metrics.*" to be more specific for metrics endpoint.

File: dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/CustomAccessLogFieldsTest.java [hunk 4]
```diff
@@ -274,10 +290,14 @@ public void testAllFieldsArePrinted() throws Exception {
...
```
Comment (fmhwong) [inline dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/CustomAccessLogFieldsTest.java:299]: Use  "SRVE0242I.*/metrics.*" to be more specific for metrics endpoint.

File: dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/CustomAccessLogFieldsTest.java [hunk 5]
```diff
@@ -330,10 +350,14 @@ public void testFieldsInAccessLogAreSameInJSON() throws Exception {
...
```
Comment (fmhwong) [inline dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/CustomAccessLogFieldsTest.java:359]: Use  "SRVE0242I.*/metrics.*" to be more specific for metrics endpoint.

File: dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/CustomAccessLogFieldsTest.java [hunk 6]
```diff
@@ -376,6 +400,12 @@ public void testFieldsInAccessLogAreSameInJSON() throws Exception {
     public void testChangeJsonAccessLogFieldsConfigValue() throws Exception {
         setUp(changeConfigServer);
         waitForSecurityPrerequisites(changeConfigServer, WAIT_TIMEOUT);
+        // Ensure defaultHttpEndpoint-ssl TCP Channel is started
+        assertNotNull("TCP Channel defaultHttpEndpoint-ssl has not started (CWWKO0219I not found)",
+                      changeConfigServer.waitForStringInLog("CWWKO0219I.*defaultHttpEndpoint-ssl", 60000));
+        // Wait for /metrics to be initialized
+        assertNotNull("/metrics was not initialized (SRVE0242I not found)",
```
Comment (fmhwong) [inline dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/CustomAccessLogFieldsTest.java:407]: Use  "SRVE0242I.*/metrics.*" to be more specific for metrics endpoint.

File: dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/CustomAccessLogFieldsTest.java [hunk 7]
```diff
@@ -583,7 +612,7 @@ private static void hitHttpEndpoint(String stringUrl, LibertyServer server) {
             assertTrue("Nothing was returned from the servlet - there was a problem connecting.", lines.length() > 0);
             con.disconnect();
         } catch (IOException e) {
-
```
Comment (fmhwong) [inline dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/CustomAccessLogFieldsTest.java:586]: Do you expect the test to fail when you catch an IOException?

File: dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/CustomAccessLogFieldsTest.java [hunk 8]
```diff
@@ -623,12 +651,11 @@ public boolean verify(String arg0, SSLSession arg1) {
             } finally {
                 br.close();
             }
-
             Log.info(c, "hitHttpsEndpointSecure", url + " reached successfully.");
             assertTrue("Nothing was returned from the servlet - there was a problem connecting.", lines.length() > 0);
             con.disconnect();
         } catch (IOException e) {
-
+            e.printStackTrace();
```
Comment (fmhwong) [inline dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/CustomAccessLogFieldsTest.java:658]: Do you expect the test to fail when you catch an IOException?

**Comment by jennifer-c** [comment]
#build

**Comment by jennifer-c** [comment]
#build

**Comment by jennifer-c** [comment]
Test on Sun EBC where it was consistently failing previously. FAT ran 10 times: https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_5GoG8LrMEeqsDL_UN9umug

**Comment by jennifer-c** [comment]
#run-libby-bot

**Pull request merged**